### PR TITLE
[Batched RPA] Port stacked scheduler to batched RPA kernel

### PR DESCRIPTION
**Pull request by weiyu0824**
# Description

Port stacked scheduler for long context support. 

# Tests

# Checklist

Before submitting this PR, please make sure:
- I have performed a self-review of my code.
- I have necessary comments in my code, particularly in hard-to-understand areas.
- I have made or will make corresponding changes to any relevant documentation.

### CODE REVIEW

File: tpu_inference/kernels/experimental/batched_rpa/wrapper.py [hunk 1]
```diff
@@ -416,46 +416,46 @@ def ragged_paged_attention(
...
-    use_causal_mask: Not used.
-
-  Returns:
-    out: [max_num_tokens, num_q_heads, head_dim]. Output of self attention.
-    new_kv_cache: [num_pages, page_size, cdiv(num_kv_heads * 2, kv_packing),
-      kv_packing, head_dim]. Result of new kv cache where k & vs are
-      concatenated along num kv heads dim.
-  """
+    Args:
+        queries: [max_num_tokens, num_q_heads, head_dim]. Output of q projection.
```
Comment (kyuyeunk) [inline tpu_inference/kernels/experimental/batched_rpa/wrapper.py:420]: can you fix indentations?

File: tpu_inference/kernels/experimental/batched_rpa/schedule.py [hunk 2]
```diff
@@ -262,32 +307,213 @@ def out_specs(self):
...
     cu_q_lens_ref: jax.Ref,
     kv_lens_ref: jax.Ref,
     distribution_ref: jax.Ref,
     schedule: RpaSchedule,
-    lane_lengths_ref: jax.Ref,
+    schedule_hbm_ref: RpaSchedule,
+    dma_sem: jax.Ref,
     *,
     cfgs: configs.RpaConfigs,
-    update_kv_cache: bool = True,
```
Comment (kyuyeunk) [inline tpu_inference/kernels/experimental/batched_rpa/schedule.py:273]: please fix this

File: tpu_inference/kernels/experimental/batched_rpa/wrapper.py (hunk 1 above)
Comment (weiyu0824) [reply tpu_inference/kernels/experimental/batched_rpa/wrapper.py:420]: this is weird, I feel like I didn't do any thing to this part.
